atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/ResilientLockWatchProxy.java

```diff
@@ -14,58 +14,66 @@
  * limitations under the License.
  */
 
-package com.palantir.atlasdb.keyvalue.api.watch;
+package com.palantir.atlasdb.keyvalue.api;
 
 import com.codahale.metrics.Counter;
 import com.google.common.reflect.AbstractInvocationHandler;
-import com.google.errorprone.annotations.concurrent.GuardedBy;
+import com.palantir.atlasdb.keyvalue.api.cache.LockWatchValueScopingCache;
 import com.palantir.atlasdb.transaction.api.TransactionLockWatchFailedException;
 import com.palantir.atlasdb.util.MetricsManager;
 import com.palantir.lock.watch.LockWatchEventCache;
 import com.palantir.logsafe.exceptions.SafeRuntimeException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.Proxy;
+import javax.annotation.concurrent.ThreadSafe;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-final class ResilientLockWatchEventCache extends AbstractInvocationHandler {
+@ThreadSafe
+public final class ResilientLockWatchProxy<T> extends AbstractInvocationHandler {
```

Comment:
The reason I didn't just wrap the merged cache with one of these is that it is conceivable that the value cache falls back (due to say, validation errors), but the event cache does not, still serving atlasdb-proxy.